train.py

```diff
@@ -399,7 +399,8 @@ def loss_fn(pred, labels):
             lr_schedulers.step()
 
             if (
-                job_config.training.enable_fp8_linear
+                SM90OrLater
```

Comment:
For a quick fix, you can have a return value `fp8_enabled` of `maybe_build_fp8_linear` and use that instead of `job_config.training.enable_fp8_linear`.